### PR TITLE
[SDK] Support "pf config set user_agent="smile-llm/0.0.1""

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.
This PR supports config user agent for 1p customer to help us distinguish usage in telemetry.

This pull request includes changes to the codebase that improve the user agent functionality and add new tests for the user agent configuration.

Main user agent changes:

* <a href="diffhunk://#diff-47208ac35b30920275fcd5e55d662647ef360129359bdc77fddd2a2157b6f47eR887-R908">`src/promptflow/promptflow/_sdk/_utils.py`</a>: Added a new class method `update_user_agent_from_config` to the `ClientUserAgentUtil` class that retrieves the user agent from the configuration and appends it to the existing user agent.
* <a href="diffhunk://#diff-ff53f9e2ad1838023fb07cafd5526a639fe9589e470f9983a7461d5d8a89f9a5R48">`src/promptflow/promptflow/_sdk/_configuration.py`</a>: Added a new constant `USER_AGENT` and a new method `get_user_agent` to the `Configuration` class for retrieving and modifying the user agent configuration value. <a href="diffhunk://#diff-ff53f9e2ad1838023fb07cafd5526a639fe9589e470f9983a7461d5d8a89f9a5R48">[1]</a> <a href="diffhunk://#diff-ff53f9e2ad1838023fb07cafd5526a639fe9589e470f9983a7461d5d8a89f9a5R218-R224">[2]</a>

Testing improvements:

* <a href="diffhunk://#diff-1a31bc7b67fad77e9c55749fba2e3305222bba2dc5d8f46766001e2e2754caaaR55-R68">`src/promptflow/tests/sdk_cli_test/unittests/test_config.py`</a>: Added a new test method to `test_config.py` for testing the behavior of setting and loading the user agent from the configuration. <a href="diffhunk://#diff-1a31bc7b67fad77e9c55749fba2e3305222bba2dc5d8f46766001e2e2754caaaR55-R68">[1]</a> <a href="diffhunk://#diff-1a31bc7b67fad77e9c55749fba2e3305222bba2dc5d8f46766001e2e2754caaaR10-R11">[2]</a>
* <a href="diffhunk://#diff-7d74ecb42a689e4b7dea2171b91f90db56f79b6efcd70629df1548a1b091f229L25-R25">`src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py`</a>: Added a new test method to `test_cli.py` for testing the behavior of setting and clearing the user agent configuration value. <a href="diffhunk://#diff-7d74ecb42a689e4b7dea2171b91f90db56f79b6efcd70629df1548a1b091f229L25-R25">[1]</a> <a href="diffhunk://#diff-7d74ecb42a689e4b7dea2171b91f90db56f79b6efcd70629df1548a1b091f229R1705-R1723">[2]</a>

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by yigao-alter — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

File: src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py [hunk 1]
```diff
@@ -1702,3 +1702,22 @@ def test_unknown_command(self, capfd):
             pass
         _, err = capfd.readouterr()
         assert "invalid choice" in err
+
+    def test_config_set_user_agent(self) -> None:
+        run_pf_command(
+            "config",
+            "set",
+            "user_agent=test/1.0.0",
```
Comment (yigao-alter) [inline src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py:1710]: user_agent = "test/1.0.0" on the PR title.
and user_agent = test/1.0.0 in this test case

Comment (D-W-) [reply src/promptflow/tests/sdk_cli_test/e2etests/test_cli.py:1710]: Offline synced, we'll keep what user specified and won't do escape for it: 
image

**Review by 0mza987 — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/_configuration.py [hunk 2]
```diff
@@ -214,3 +215,10 @@ def _validate(key: str, value: str) -> None:
                     "please use its child folder, e.g. '${flow_directory}/.runs'."
                 )
         return
+
+    def get_user_agent(self) -> Optional[str]:
+        """Get 1p customer set user agent. Only 1p customer will set this. If set, will add prefix `PFCustomer_`"""
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_configuration.py:220]: don't mention 1p in source code

Comment (D-W-) [reply src/promptflow/promptflow/_sdk/_configuration.py]: Updated

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**